node-manager/src/localstorage.rs

```diff
@@ -0,0 +1,952 @@
+                val.transaction = raw_tx;
+
+                Ok(Some(val))
+            }
+        }
+    }
+    fn del_last_index(&mut self, keychain: KeychainKind) -> Result<Option<u32>, bdk::Error> {
+        let key = MapKey::LastIndex(keychain).as_map_key();
+        let res: Option<u32> = LocalStorage::get(&key).ok();
+        LocalStorage::delete(&key);
+
+        Ok(res)
+    }
+    fn del_sync_time(&mut self) -> Result<Option<SyncTime>, bdk::Error> {
+        let key = MapKey::SyncTime.as_map_key();
+        let res: Option<SyncTime> = LocalStorage::get(&key).ok();
+        LocalStorage::delete(&key);
+
+        Ok(res)
+    }
+}
+
+impl Database for MutinyBrowserStorage {
+    fn check_descriptor_checksum<B: AsRef<[u8]>>(
+        &mut self,
+        keychain: KeychainKind,
+        bytes: B,
+    ) -> Result<(), bdk::Error> {
+        let key = MapKey::DescriptorChecksum(keychain).as_map_key();
+
+        let prev = LocalStorage::get::<Vec<u8>>(&key).ok();
+        if let Some(val) = prev {
+            if val == bytes.as_ref().to_vec() {
+                Ok(())
+            } else {
+                Err(bdk::Error::ChecksumMismatch)
+            }
+        } else {
+            self.set(key, bytes.as_ref().to_vec())?;
+            Ok(())
+        }
+    }
+
+    fn iter_script_pubkeys(
+        &self,
+        keychain: Option<KeychainKind>,
+    ) -> Result<Vec<Script>, bdk::Error> {
+        let key = MapKey::Path((keychain, None)).as_map_key();
+        self.scan_prefix(key)
+            .into_iter()
+            .map(|tuple| -> Result<_, bdk::Error> {
+                // todo make this safer
+                let (_, value) = tuple;
+                let str = value.as_str().unwrap();
+                Script::from_hex(str)
+                    .map_err(|_| bdk::Error::Generic(String::from("Error decoding")))
+            })
+            .collect()
+    }
+
+    fn iter_utxos(&self) -> Result<Vec<LocalUtxo>, bdk::Error> {
+        let key = MapKey::Utxo(None).as_map_key();
+        self.scan_prefix(key)
+            .into_iter()
+            .map(|x| -> Result<_, bdk::Error> {
+                let (_, v) = x;
+
+                let utxo: LocalUtxo = Deserialize::deserialize(v)?;
+                Ok(utxo)
+            })
+            .collect()
+    }
+
+    fn iter_raw_txs(&self) -> Result<Vec<Transaction>, bdk::Error> {
+        let key = MapKey::RawTx(None).as_map_key();
+        self.scan_prefix(key)
+            .into_iter()
+            .map(|tuple| -> Result<_, bdk::Error> {
+                let (_, value) = tuple;
+                let tx: Transaction = Deserialize::deserialize(value)?;
+                Ok(tx)
+            })
+            .collect()
+    }
+
+    fn iter_txs(&self, include_raw: bool) -> Result<Vec<TransactionDetails>, bdk::Error> {
+        let key = MapKey::Transaction(None).as_map_key();
+        self.scan_prefix(key)
+            .into_iter()
+            .map(|tuple| -> Result<_, bdk::Error> {
```

Comment:
you can destructure this right inside the `||`:

|(key, value)| ->

no need for the intermediate `tuple`